pkg/osutil/osutil.go

```diff
@@ -115,6 +123,14 @@ func CopyDir(
 	return nil
 }
 
+func validateAllVariablesSubstituted(s string) error {
+	if unsubstitutedVars := draftVariableRegex.FindAllString(s, -1); len(unsubstitutedVars) > 0 {
+		unsubstitutedVarsString := strings.Join(unsubstitutedVars, ", ")
+		return fmt.Errorf("unsubstituted variable: %s", unsubstitutedVarsString)
```

Comment:
💡 What is the contract of this funct?

* I created a playground for better understanding: https://go.dev/play/p/SSGWyrp5rB6

So, given you will have no spaces and curleys you will error out? otherwise not?

Reply:
Excluding helm logic syntax from breaking replacement variable checks

Reply:
if you have a variable with no spaces and with curleys around it then yes, it will error since that means that a variable was missed in the draft substitution.
i can further constrain this contract to exclude values in curlys that start with a period since that is only a helm thing too